src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,175 @@ void testGetStringValues() throws Exception {
+        assertNull(new MongoSQLBsonValue(new BsonUndefined()).toString());
+        assertNull(mongoSQLResultSetAllTypes.getObject(ALL_NULL_COL_LABEL));
+        assertNull(new MongoSQLBsonValue(new BsonNull()).toString());
+
+        // Note that getObject() must return the following classes for the following types:
+        //   - Types.DECIMAL   => java.math.BigDecimal
+        //   - Types.BINARY    => byte[]
+        //   - Types.TIMESTAMP => java.sql.Timestamp
+        // Therefore, the getObject().toString() representations are not extended JSON.
+        // Since Types.BINARY maps to an array, we omit its getObject().toString() test.
+        assertEquals("21.2", mongoSQLResultSetAllTypes.getObject(ALL_DECIMAL_COL_LABEL).toString());
+
+        Codec<BsonDateTime> c = new BsonDateTimeCodec();
+        BsonDateTime d =
+                c.decode(new JsonReader(ALL_DATE_COL_VAL), DecoderContext.builder().build());
+        Timestamp t = new Timestamp(d.getValue());
+
+        assertEquals(
+                t.toString(), mongoSQLResultSetAllTypes.getObject(ALL_DATE_COL_LABEL).toString());
+    }
+
+    @Test
+    public void testGetObjectToStringMatchesGetString() throws Exception {
+        // Assert that getObject().toString() matches getString() for BSON types
+        // that map to JDBC Types.OTHER.
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_OBJECT_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_OBJECT_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_ARRAY_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_ARRAY_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_OBJECT_ID_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_OBJECT_ID_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_REGEX_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_REGEX_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_DB_POINTER_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_DB_POINTER_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_JAVASCRIPT_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_SYMBOL_COL_LABEL),
+                mongoSQLResultSetAllTypes.getObject(ALL_SYMBOL_COL_LABEL).toString());
+        assertEquals(
+                mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL),
+                mongoSQLResultSetAllTypes
+                        .getObject(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL)
+                        .toString());
+        assertEquals(
```

Comment:
Same here, I am not sure how this assert passes since getString should return {"$timestamp": {"t": 1412180887, "i": 1}} and getObject.toString should return something like "2020-12-25 12:13:14.0" (value formatted with yyyy-mm-dd hh:mm:ss.fffffffff)

Reply:
Got it, thanks Matt for the extra details on the fun of Bson.Timestamp :)

Reply:
discussed offline: bson timestamp != bson date